nntrainer/layers/split_layer.cpp

```diff
@@ -133,12 +135,14 @@ void SplitLayer::calcDerivative(RunLayerContext &context) {
     output_.reshape(output_reshape_helper);
 
     for (unsigned int batch = 0; batch < input_.batch(); batch++) {
-      Tensor dest_tensor = Tensor::Map(input_.getAddress(batch, 0, idx, 0),
-                                       input_reshape_helper.width(),
-                                       {1, 1, 1, input_reshape_helper.width()});
-      const Tensor source_tensor = Tensor::Map(
-        output_.getAddress(batch, 0, 0, 0), output_reshape_helper.width(),
-        {1, 1, 1, output_reshape_helper.width()});
+      Tensor dest_tensor =
+        Tensor::Map(input_.getAddress(batch, 0, idx, 0),
+                    input_reshape_helper.width() * sizeof(float),
+                    {1, 1, 1, input_reshape_helper.width()});
+      const Tensor source_tensor =
+        Tensor::Map(output_.getAddress(batch, 0, 0, 0),
+                    output_reshape_helper.width() * sizeof(float),
```

Comment:
This works but as Tensor only deals with float, you can also consider updating Map to take offset as elements than bytes as it's a bit overhead to calculate bytes everywhere.

Reply:
I think it's better to make the offset as elements, this is more like a bugfix rather then an enhancement, so let's deal this in different PR.

Reply:
we can have a tensor size type defined as a template meta function. using decltype. I'll do this as a separate PR.